src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
```

Comment:
The method `convertToTransaction` seems to be doing too much. It's parsing the data, validating it, and also creating `FruitTransaction` objects. Consider applying the Single Responsibility Principle (SRP) from SOLID and break this method down to adhere to SRP.